### PR TITLE
python311Packages.dissect: 3.5 -> 3.6 

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
Bump dissect stack.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 234492` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.apache-beam</li>
    <li>python311Packages.apache-beam.dist</li>
  </ul>
</details>
<details>
  <summary>126 packages built:</summary>
  <ul>
    <li>acquire (python310Packages.acquire)</li>
    <li>acquire.dist (python310Packages.acquire.dist)</li>
    <li>python310Packages.apache-beam</li>
    <li>python310Packages.apache-beam.dist</li>
    <li>python310Packages.dissect</li>
    <li>python310Packages.dissect-cim</li>
    <li>python310Packages.dissect-cim.dist</li>
    <li>python310Packages.dissect-clfs</li>
    <li>python310Packages.dissect-clfs.dist</li>
    <li>python310Packages.dissect-cobaltstrike</li>
    <li>python310Packages.dissect-cobaltstrike.dist</li>
    <li>python310Packages.dissect-cstruct</li>
    <li>python310Packages.dissect-cstruct.dist</li>
    <li>python310Packages.dissect-esedb</li>
    <li>python310Packages.dissect-esedb.dist</li>
    <li>python310Packages.dissect-etl</li>
    <li>python310Packages.dissect-etl.dist</li>
    <li>python310Packages.dissect-eventlog</li>
    <li>python310Packages.dissect-eventlog.dist</li>
    <li>python310Packages.dissect-evidence</li>
    <li>python310Packages.dissect-evidence.dist</li>
    <li>python310Packages.dissect-executable</li>
    <li>python310Packages.dissect-executable.dist</li>
    <li>python310Packages.dissect-extfs</li>
    <li>python310Packages.dissect-extfs.dist</li>
    <li>python310Packages.dissect-fat</li>
    <li>python310Packages.dissect-fat.dist</li>
    <li>python310Packages.dissect-ffs</li>
    <li>python310Packages.dissect-ffs.dist</li>
    <li>python310Packages.dissect-hypervisor</li>
    <li>python310Packages.dissect-hypervisor.dist</li>
    <li>python310Packages.dissect-ntfs</li>
    <li>python310Packages.dissect-ntfs.dist</li>
    <li>python310Packages.dissect-ole</li>
    <li>python310Packages.dissect-ole.dist</li>
    <li>python310Packages.dissect-regf</li>
    <li>python310Packages.dissect-regf.dist</li>
    <li>python310Packages.dissect-shellitem</li>
    <li>python310Packages.dissect-shellitem.dist</li>
    <li>python310Packages.dissect-sql</li>
    <li>python310Packages.dissect-sql.dist</li>
    <li>python310Packages.dissect-squashfs</li>
    <li>python310Packages.dissect-squashfs.dist</li>
    <li>python310Packages.dissect-target</li>
    <li>python310Packages.dissect-target.dist</li>
    <li>python310Packages.dissect-thumbcache</li>
    <li>python310Packages.dissect-thumbcache.dist</li>
    <li>python310Packages.dissect-util</li>
    <li>python310Packages.dissect-util.dist</li>
    <li>python310Packages.dissect-vmfs</li>
    <li>python310Packages.dissect-vmfs.dist</li>
    <li>python310Packages.dissect-volume</li>
    <li>python310Packages.dissect-volume.dist</li>
    <li>python310Packages.dissect-xfs</li>
    <li>python310Packages.dissect-xfs.dist</li>
    <li>python310Packages.dissect.dist</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.fastavro</li>
    <li>python310Packages.fastavro.dist</li>
    <li>python310Packages.flow-record</li>
    <li>python310Packages.flow-record.dist</li>
    <li>python310Packages.hdfs</li>
    <li>python310Packages.hdfs.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.acquire</li>
    <li>python311Packages.acquire.dist</li>
    <li>python311Packages.dissect</li>
    <li>python311Packages.dissect-cim</li>
    <li>python311Packages.dissect-cim.dist</li>
    <li>python311Packages.dissect-clfs</li>
    <li>python311Packages.dissect-clfs.dist</li>
    <li>python311Packages.dissect-cobaltstrike</li>
    <li>python311Packages.dissect-cobaltstrike.dist</li>
    <li>python311Packages.dissect-cstruct</li>
    <li>python311Packages.dissect-cstruct.dist</li>
    <li>python311Packages.dissect-esedb</li>
    <li>python311Packages.dissect-esedb.dist</li>
    <li>python311Packages.dissect-etl</li>
    <li>python311Packages.dissect-etl.dist</li>
    <li>python311Packages.dissect-eventlog</li>
    <li>python311Packages.dissect-eventlog.dist</li>
    <li>python311Packages.dissect-evidence</li>
    <li>python311Packages.dissect-evidence.dist</li>
    <li>python311Packages.dissect-executable</li>
    <li>python311Packages.dissect-executable.dist</li>
    <li>python311Packages.dissect-extfs</li>
    <li>python311Packages.dissect-extfs.dist</li>
    <li>python311Packages.dissect-fat</li>
    <li>python311Packages.dissect-fat.dist</li>
    <li>python311Packages.dissect-ffs</li>
    <li>python311Packages.dissect-ffs.dist</li>
    <li>python311Packages.dissect-hypervisor</li>
    <li>python311Packages.dissect-hypervisor.dist</li>
    <li>python311Packages.dissect-ntfs</li>
    <li>python311Packages.dissect-ntfs.dist</li>
    <li>python311Packages.dissect-ole</li>
    <li>python311Packages.dissect-ole.dist</li>
    <li>python311Packages.dissect-regf</li>
    <li>python311Packages.dissect-regf.dist</li>
    <li>python311Packages.dissect-shellitem</li>
    <li>python311Packages.dissect-shellitem.dist</li>
    <li>python311Packages.dissect-sql</li>
    <li>python311Packages.dissect-sql.dist</li>
    <li>python311Packages.dissect-squashfs</li>
    <li>python311Packages.dissect-squashfs.dist</li>
    <li>python311Packages.dissect-target</li>
    <li>python311Packages.dissect-target.dist</li>
    <li>python311Packages.dissect-thumbcache</li>
    <li>python311Packages.dissect-thumbcache.dist</li>
    <li>python311Packages.dissect-util</li>
    <li>python311Packages.dissect-util.dist</li>
    <li>python311Packages.dissect-vmfs</li>
    <li>python311Packages.dissect-vmfs.dist</li>
    <li>python311Packages.dissect-volume</li>
    <li>python311Packages.dissect-volume.dist</li>
    <li>python311Packages.dissect-xfs</li>
    <li>python311Packages.dissect-xfs.dist</li>
    <li>python311Packages.dissect.dist</li>
    <li>python311Packages.fastavro</li>
    <li>python311Packages.fastavro.dist</li>
    <li>python311Packages.flow-record</li>
    <li>python311Packages.flow-record.dist</li>
    <li>python311Packages.hdfs</li>
    <li>python311Packages.hdfs.dist</li>
  </ul>
</details>

**Comment by samuela** [comment]
`python311Packages.apache-beam` is failing on master according to hydra

File: pkgs/development/python-modules/fastavro/default.nix [hunk 1]
```diff
@@ -1,8 +1,8 @@
-{ buildPythonPackage
+{ lib
+, buildPythonPackage
 , cython
```
Comment (samuela) [inline pkgs/development/python-modules/fastavro/default.nix:3]: please keep alphabetical order

File: pkgs/development/python-modules/fastavro/default.nix [hunk 2]
```diff
@@ -15,44 +15,69 @@
...
   };
 
   preBuild = ''
     export FASTAVRO_USE_CYTHON=1
   '';
 
-  nativeBuildInputs = [ cython ];
+  nativeBuildInputs = [
+    cython
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/fastavro/default.nix:36]: let's keep this as is to minimize polluting the git blame

File: pkgs/development/python-modules/fastavro/default.nix [hunk 3]
```diff
@@ -15,44 +15,69 @@
...
   # CLI tests are broken on Python 3.8. See https://github.com/fastavro/fastavro/issues/558.
-  disabledTestPaths = lib.optionals isPy38 [ "tests/test_main_cli.py" ];
+  disabledTestPaths = lib.optionals isPy38 [
+    "tests/test_main_cli.py"
+  ];
 
-  pythonImportsCheck = [ "fastavro" ];
+  pythonImportsCheck = [
+    "fastavro"
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/fastavro/default.nix:75]: again, please leave as is to minimize polluting the git blame

**Review by samuela — DISMISSED** [review]

File: pkgs/development/python-modules/fastavro/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/fastavro/default.nix:3]: `lib` is located first for the vast majority of Python modules and the position where `nix-init` puts it.

Comment (samuela) [reply pkgs/development/python-modules/fastavro/default.nix:3]: I know it's a matter of style... it's just that my style is to keep things alphabetical :P this change is also not relevant to the real content of the PR, better to keep the git blame clean

**Review by samuela — APPROVED** [review]
Fastavro diff LGTM!

**Pull request merged**